### PR TITLE
Fix mongodb processor error handling

### PR DESCRIPTION
**Pull request by mihaitodor**

### CODE REVIEW

File: internal/impl/mongodb/processor.go [hunk 1]
```diff
@@ -228,8 +228,13 @@ func NewProcessor(conf processor.Config, mgr bundle.NewManagement) (processor.V2
...
 func (m *Processor) ProcessBatch(ctx context.Context, spans []*tracing.Span, batch message.Batch) ([]message.Batch, error) {
 	writeModelsMap := map[*mongo.Collection][]mongo.WriteModel{}
-	_ = batch.Iter(func(i int, p *message.Part) error {
-		var err error
+	_ = batch.Iter(func(i int, p *message.Part) (err error) {
+		defer func() {
+			if err != nil {
+				p.ErrorSet(err)
+			}
+		}()
```
Comment (mihaitodor) [inline internal/impl/mongodb/processor.go:236]: @lucasoares this should fix the broken integration test

File: internal/integration/stream_test_helpers.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 // specifically requested using the -run flag.
 func CheckSkip(t testing.TB) {
 	if m := flag.Lookup("test.run").Value.String(); m == "" || regexp.MustCompile(strings.Split(m, "/")[0]).FindString(t.Name()) == "" {
-		t.Skip("Skipping as execution was not requested explicitly using go test -run ^TestIntegration$")
+		t.Skip("Skipping as execution was not requested explicitly using go test -run ^Test.*Integration.*$")
```
Comment (mihaitodor) [inline internal/integration/stream_test_helpers.go:36]: Thought this should reflect whatever's in the Makefile, so it matches all integration test variations.

File: internal/impl/mongodb/processor.go (hunk 1 above)
Comment (lucasoares) [reply internal/impl/mongodb/processor.go:236]: Nice!

**Comment by lucasoares** [comment]
Seems OK to me :+1:

**Review by Jeffail — APPROVED** [review]
LGTM, extra ❤️ for clearing up for integration test flag checks.

**Pull request merged**